sdk/packages/selector-solana/src/adapter.ts

```diff
@@ -0,0 +1,479 @@
+
+  private _connecting: boolean
+  private _connected: boolean
+  private _publicKey: PublicKey | null
+  private _readyState: WalletReadyState =
+    typeof window === 'undefined' || typeof document === 'undefined'
+      ? WalletReadyState.Unsupported
+      : WalletReadyState.NotDetected
+
+  private _app: AppSolana | undefined
+  private _appSessionActive: boolean
+  private _innerStandardAdapter: StandardWalletAdapter | undefined
+  private _modal: NightlyConnectSelectorModal | undefined
+
+  private _appInitData: AppInitData
+  private _eagerConnectForStandardWallets: boolean
+
+  private _walletsList: IWalletListItem[] = []
+
+  private _chosenMobileWalletName: string | undefined
+
+  constructor(appInitData: AppInitData, eagerConnectForStandardWallets?: boolean) {
+    super()
+    this._connecting = false
+    this._connected = false
+    this._publicKey = null
+    this._appInitData = appInitData
+    this._eagerConnectForStandardWallets = !!eagerConnectForStandardWallets
+    this._appSessionActive = false
+  }
+
+  get connecting() {
+    return this._connecting
+  }
+
+  get connected() {
+    return this._connected
+  }
+
+  get readyState() {
+    return this._readyState
+  }
+
+  get publicKey() {
+    return this._publicKey
+  }
+
+  public static async build(
+    appInitData: AppInitData,
+    eagerConnectForStandardWallets?: boolean,
+    anchorRef?: HTMLElement
+  ) {
+    const adapter = new NightlyConnectAdapter(appInitData, eagerConnectForStandardWallets)
+
+    const [app, metadataWallets] = await Promise.all([
+      AppSolana.build(appInitData),
+      AppSolana.getWalletsMetadata('https://nc2.nightly.app/get_wallets_metadata')
+        .then((list) =>
+          list.map((wallet) => ({
+            name: wallet.name,
+            icon: wallet.image.default,
+            deeplink: wallet.mobile,
+            link: wallet.homepage
+          }))
+        )
+        .catch(() => [] as MetadataWallet[])
+    ])
+
+    adapter._app = app
+
+    adapter._walletsList = getWalletsList(
+      metadataWallets,
+      solanaWalletsFilter,
+      getRecentStandardWalletForNetwork(SOLANA_NETWORK) ?? undefined
+    )
+
+    adapter._modal = new NightlyConnectSelectorModal(
+      adapter._walletsList,
+      appInitData.url ?? 'https://nc2.nightly.app',
+      {
+        network: QueryNetwork.SOLANA,
+        name: SOLANA_NETWORK,
+        icon: 'https://assets.coingecko.com/coins/images/4128/small/solana.png'
+      },
+      anchorRef,
+      undefined,
+      () => {
+        adapter._connecting = false
+      }
+    )
+
+    adapter._readyState = WalletReadyState.Installed
+
+    return adapter
+  }
+
+  public static buildLazy(
+    appInitData: AppInitData,
+    eagerConnectForStandardWallets?: boolean,
+    anchorRef?: HTMLElement
+  ) {
+    const adapter = new NightlyConnectAdapter(appInitData, eagerConnectForStandardWallets)
+
+    Promise.all([
+      AppSolana.build(appInitData),
```

Comment:
you should set wallet readystate on loading here

Reply:
I don't understand why, you have told me that we should set ready state after app build; I fear that if I set this before app is built, there will be problems with things like eager connect, etc.

Reply:
fixed